packages/medusa/src/loaders/medusa-app.ts

```diff
@@ -0,0 +1,58 @@
+import { MedusaApp, ModulesDefinition } from "@medusajs/modules-sdk"
+
+import { CommonTypes } from "@medusajs/types"
+import { ContainerRegistrationKeys } from "@medusajs/utils"
+import { asValue } from "awilix"
+import { joinerConfig } from "../joiner-config"
+import { mergeModulesConfig } from "../utils/merge-modules-config"
+import modulesConfig from "../modules-config"
+import { remoteQueryFetchData } from ".."
+
+export const loadMedusaApp = async (
+  {
+    configModule,
+    container,
+  }: { configModule: CommonTypes.ConfigModule; container: any },
+  config = { register: true }
+) => {
+  mergeModulesConfig(configModule.modules ?? {}, modulesConfig)
+
+  const injectedDependencies = {
+  }
+
+  const sharedResourcesConfig = {
+    database: {
+      clientUrl: configModule.projectConfig.database_url,
+    },
+  }
+
+  const medusaApp = await MedusaApp({
+    modulesConfig,
+    servicesConfig: joinerConfig,
+    remoteFetchData: remoteQueryFetchData(container),
+    sharedResourcesConfig, 
+    injectedDependencies
+  })
+
+  if (!config.register) {
+    return medusaApp
+  }
+
+  container.register("medusaApp", asValue(medusaApp))
```

Comment:
I wouldn't register all of that in the container. But instead return the return from the index loader such as we do for the container. Than you can use it just like we do for the container when calling the bootstrap

Reply:
we need at least the link modules when we get to the workflow handlers, how would you prefer handling this, registering just the link service?